tests/datasource/fluent/data_asset/test_sql_asset.py

```diff
@@ -168,3 +232,83 @@ def test_add_batch_definition_fluent_sql__add_batch_definition_daily(
     # assert
     assert batch_definition == expected_batch_definition
     datasource.add_batch_definition.assert_called_once_with(expected_batch_definition)
+
+
+class ExpectNoError:
+    def __enter__(self): ...
+
+    def __exit__(self, exc_type, exc_val, exc_tb): ...
+
+
+def add_table_asset(datasource: SQLDatasource):
+    return datasource.add_table_asset(
+        name="my_table_asset",
+        table_name="my_table",
+        schema_name="my_schema",
+    )
+
+
+def add_query_asset(datasource: SQLDatasource):
+    return datasource.add_query_asset(name="my_query_asset", query="select * from my_table")
+
+
+@pytest.mark.unit
```

Comment:
This is the test that exercises this new check. The other test changes are updating previous tests now that we may actually connect to a db.